### ISSUE TITLE
Changing `dt` between runs is not working

### ISSUE THREAD

**Issue by denisalevi** [brian-team/brian2cuda#136]
Implement [this brian2 test](https://github.com/brian-team/brian2/blob/320a5f6b36e2aa5dfe7e6b22bd46c3652ad9cc99/brian2/tests/test_synapses.py#L1084-L1106) for brian2cuda. A first test did not show any spikes in cuda, but some in cpp (but also not all that are tested for).

I just came across  [this section in the `CPPSpikeQueue`](https://github.com/brian-team/brian2/blob/320a5f6b36e2aa5dfe7e6b22bd46c3652ad9cc99/brian2/synapses/cspikequeue.cpp#L58-L74), where in case of a changed `dt` the spikes in the queues are sorted into new queues. I don't think we are doing anything like this at all?

**Comment by moritzaugustin** [comment]
I don't think we necessarily should support variable time steps in brian2cuda

@mstimberg do you think differently?

**Comment by denisalevi** [comment]
This is actually related to #86. We are just rerunning the `synapses_initialise_queue` template as soon as any synapse variable (I guess?) is changed between run calls (maybe same for `dt`?). We will have take care of that anyways and while doing it, we can copy the code I linked above from the `cspikequeue.cpp`.

**Comment by mstimberg** [comment]
> I don't think we necessarily should support variable time steps in brian2cuda

I agree that this is a bit of a niche feature that I did not come across "in the wild" yet. The only thing that is important to me (this was an improvement over Brian 1) is that `dt` should be changeable until the `run` call -- i.e. if a user specifies `defaultclock.dt = ...` not at the start of the script but later, it shouldn't screw up things. I would not mind at all throwing a `NotImplementedError` if the `dt` changes between runs.

**Comment by mstimberg** [comment]
Hmm, this seems indeed to be a bug in Brian. For some reason, we never made this test "standalone-compatible", so it was only running for the runtime targets (where it obviously works). Maybe as a short-term solution, one could just throw an error if `dt` changes between runs in standalone mode (at least when it affects the spike queue)?
